src/parcsr_ls/par_mgr.c

```diff
@@ -3060,19 +3081,54 @@ hypre_MGRBuildInterp(hypre_ParCSRMatrix   *A,
   }
   else if (interp_type == 4)
   {
-    hypre_MGRBuildInterpApproximateInverse(A, CF_marker, num_cpts_global, debug_flag, &P_ptr);
-    hypre_BoomerAMGInterpTruncation(P_ptr, trunc_factor, max_elmts);
+    if (exec == HYPRE_EXEC_HOST)
+    {
+       hypre_MGRBuildInterpApproximateInverse(A, CF_marker, num_cpts_global, debug_flag, &P_ptr);
+       hypre_BoomerAMGInterpTruncation(P_ptr, trunc_factor, max_elmts);
+    }
+#if defined(HYPRE_USING_CUDA)
```

Comment:
Also `|| defined(HYPRE_USING_HIP)`? Not sure if MGR support HIP yet.

Reply:
Thanks @liruipeng